libmamba/src/core/fetch.cpp

```diff
@@ -260,18 +260,34 @@ namespace mamba
             curl_easy_setopt(handle, CURLOPT_SSL_OPTIONS, CURLSSLOPT_NO_REVOKE);
         }
 
+        std::optional<std::string> proxy = proxy_match(url);
+        if (proxy)
+        {
+            curl_easy_setopt(handle, CURLOPT_PROXY, proxy->c_str());
+            LOG_INFO << "Using Proxy " << *proxy;
```

Comment:
Assuming that this may contain authentication info, we should strip it before logging. Not sure how it is implemented elsewhere.